extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -20,285 +20,251 @@ export async function createAzureFunction(node?: ITreeNodeInfo): Promise<void> {
 	// telemetry properties for create azure function
 	let sessionId: string = uuid.v4();
 	let propertyBag: { [key: string]: string } = { sessionId: sessionId };
-	let quickPickStep: string = '';
-	let exitReason: string = 'cancelled';
+	let telemetryStep: string = '';
+	let exitReason: string = ExitReason.cancelled;
 	TelemetryReporter.sendActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.startCreateAzureFunctionWithSqlBinding);
-
-	let selectedBindingType: BindingType | undefined;
 	let connectionInfo: IConnectionInfo | undefined;
-	let connectionURI: string;
-	let listDatabases: string[] | undefined;
-	let objectName: string | undefined;
-	const vscodeMssqlApi = await utils.getVscodeMssqlApi();
-	if (!node) {
-		// if user selects command in command palette we prompt user for information
-		quickPickStep = 'launchFromCommandPalette';
-		try {
-			// Ask binding type for promptObjectName
-			quickPickStep = 'getBindingType';
-			let selectedBinding = await azureFunctionsUtils.promptForBindingType();
-			if (!selectedBinding) {
-				return;
-			}
-			selectedBindingType = selectedBinding;
-			propertyBag.bindingType = selectedBindingType;
-			TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.startCreateAzureFunctionWithSqlBinding)
+	let isCreateNewProject: boolean = false;
+	let newFunctionFileObject: azureFunctionsUtils.IFileFunctionObject | undefined;
+
+	try {
+		const azureFunctionApi = await azureFunctionsUtils.getAzureFunctionsExtensionApi();
+		if (!azureFunctionApi) {
+			propertyBag.exitReason = exitReason;
+			TelemetryReporter.createErrorEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.exitCreateAzureFunctionQuickpick)
 				.withAdditionalProperties(propertyBag).send();
+			return;
+		}
+		/**
+		 * Prompts user for azure function project path to use
+		 * If multiple found in workspace we prompt user to pick one
+		 * If one found in workspace we use that
+		 * If none found in workspace we show error message but continue with createFunction message
+		 */
+		let projectFile = await azureFunctionsUtils.getAzureFunctionProject();
+		let projectFolder: string;
+		telemetryStep = CreateAzureFunctionStep.getAzureFunctionProject;
+		if (!projectFile) {
+			while (true) {
+				// show warning message that user needs an azure function project to create a function
+				let projectCreate = await vscode.window.showErrorMessage(constants.azureFunctionsProjectMustBeOpened,
+					constants.createProject, constants.learnMore);
+				if (projectCreate === constants.learnMore) {
+					telemetryStep = CreateAzureFunctionStep.learnMore;
+					void vscode.commands.executeCommand('vscode.open', vscode.Uri.parse(constants.sqlBindingsDoc));
+					TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, telemetryStep)
```

Comment:
Do you need to send this event? Couldn't you just use the exit event and check for step == learnMore?

Reply:
Yeah, that's right. We could also use the final block to see if it exited with the step learnMore.